risc0/circuit/rv32im/src/execute/executor.rs

```diff
@@ -326,37 +296,20 @@ impl<'a, 'b, S: Syscall> Executor<'a, 'b, S> {
 
             let final_cycles = self.segment_cycles().next_power_of_two();
             let final_po2 = log2_ceil(final_cycles as usize);
-            let partial_image = self.pager.commit();
-            let req = CreateSegmentRequest {
-                partial_image,
-                page_indexes: self.pager.page_indexes(),
-                input_digest: self.input_digest,
-                output_digest: self.output_digest,
-                read_record: std::mem::take(&mut self.read_record),
-                write_record: std::mem::take(&mut self.write_record),
-                user_cycles: self.user_cycles,
-                pager_cycles: self.pager.cycles,
-                terminate_state: self.terminate_state,
-                segment_threshold: 0, // meaningless for final segment
-                po2: final_po2 as u32,
-                index: segment_counter as u64,
-                dump_path: None,
-                povw_nonce: self.povw_nonce(segment_counter),
-            };
-            if commit_sender.send(req).is_err() {
-                return Err(segment_callback_thread.join().unwrap().unwrap_err());
-            }
-
-            let final_cycles = final_cycles as u64;
-            let user_cycles = self.user_cycles as u64;
-            let pager_cycles = self.pager.cycles as u64;
-            self.cycles.total += final_cycles;
-            self.cycles.paging += pager_cycles;
-            self.cycles.reserved += final_cycles - pager_cycles - user_cycles;
+            self.split_segment(
+                &commit_sender,
+                &mut segment_callback_thread,
+                final_po2,
+                0, // theshold is meaningless for final segment
+            )?;
 
             drop(commit_sender);
 
-            segment_callback_thread.join().unwrap()
+            segment_callback_thread
+                .take()
+                .expect("segment_callback_thread is None")
+                .join()
+                .expect("segment_callback_thread panicked")
```

Comment:
the message passed into `expect` is suppose to be an expectation, not what went wrong. I like to think about "should" statements like `"segment_callback_thread should be Some(..)"`

https://doc.rust-lang.org/stable/std/error/index.html#common-message-styles

Reply:
Ya, this is one I don't always follow. I have a feeling when I read these messages, espcially when something actually goes wrong, of "Ok, sure I know what _should have_ happened. What actually happened?". I adjusted this though. The second one should simply be an unwrap I think, since we are just propagating the `panic`

50e9a5ec8af600762b6e515baff26ccdd05577e9